api/src/main/java/org/openmrs/module/registrationcore/api/impl/PatientCreatedListener.java

```diff
@@ -0,0 +1,129 @@
+package org.openmrs.module.registrationcore.api.impl;
+
+import javax.jms.Message;
+import org.apache.commons.lang.exception.ExceptionUtils;
+import org.openmrs.Patient;
+import org.openmrs.PatientIdentifier;
+import org.openmrs.PatientIdentifierType;
+import org.openmrs.User;
+import org.openmrs.event.Event;
+import org.openmrs.module.registrationcore.api.errorhandling.ErrorHandlingService;
+import org.openmrs.module.registrationcore.api.errorhandling.PixErrorHandlingService;
+import org.openmrs.module.registrationcore.api.mpi.common.MpiException;
+import org.openmrs.module.registrationcore.api.mpi.common.MpiProperties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class listens for patient CREATED events.
+ * If MPI is enabled, it exports patient to MPI,
+ * and updates the local patient with the new patient identifier generated by the MPI.
+ */
+public class PatientCreatedListener extends PatientActionListener {
+    
+    private static final Logger LOGGER = LoggerFactory.getLogger(PatientCreatedListener.class);
+    
+    private MpiProperties mpiProperties;
+
+    private IdentifierBuilder identifierBuilder;
+
+    public void setMpiProperties(MpiProperties mpiProperties) {
+        this.mpiProperties = mpiProperties;
+    }
+
+    public void setIdentifierBuilder(IdentifierBuilder identifierBuilder) {
+        this.identifierBuilder = identifierBuilder;
+    }
+
+    /**
+     * Defines the list of Actions that this subscribable event listener class listens out for.
+     *
+     * @return a list of Actions this listener can deal with
+     */
+    public List<String> subscribeToActions(){
+        List actions = new ArrayList<String>();
+        actions.add(Event.Action.CREATED.name());
+        return actions;
+    }
+
+    /**
+     * Exports patient to MPI, and updates the local patient
+     * with the new patient identifier generated by the MPI.
+     *
+     * @param message message with properties.
+     */
+    @Override
+    public void performMpiAction(Message message) {
+        Patient patient = extractPatient(message);
+        if (patient != null){
+            PatientIdentifierType mpiPatientIdType = patientService.getPatientIdentifierTypeByUuid(mpiProperties.getMpiPersonIdentifierTypeUuid());
+            PatientIdentifier mpiPatientId = patient.getPatientIdentifier(mpiPatientIdType);
+            if (mpiPatientId != null && !mpiPatientId.getIdentifier().isEmpty()) {
```

Comment:
Use apache commons StringUtils.isNotBlank(), otherwise isEmpty() might not take care of null and white space characters

Reply:
Changed, and also changed in other places where I used isEmpty() on strings